aspnetcore/fundamentals/startup.md

```diff
@@ -14,82 +14,80 @@ uid: fundamentals/startup
+
+The `Startup` class constructor can accept dependencies that are provided through [dependency injection](dependency-injection.md). You can use `IHostingEnvironment` to set up [configuration](configuration.md) sources and `ILoggerFactory` to set up [logging](logging.md) providers. 
 
-The `Startup` class can optionally accept dependencies in its constructor that are provided through [dependency injection](dependency-injection.md).  Typically, the way an application will be configured is defined within its Startup class's constructor (see [Configuration](configuration.md)). The Startup class must define a `Configure` method, and may optionally also define a `ConfigureServices` method, which will be called when the application is started.
+The `Startup` class must include a `Configure` method and can optionally include a `ConfigureServices` method, both of which are called when the application starts. The class can also include [environment-specific versions of these methods](environments.md#startup-conventions).
 
 ## The Configure method
 
-The `Configure` method is used to specify how the ASP.NET application will respond to individual HTTP requests. At its simplest, you can configure every request to receive the same response. However, most real-world applications require more functionality than this. More complex sets of pipeline configuration can be encapsulated in [middleware](middleware.md) and added using extension methods on `IApplicationBuilder`.
+The `Configure` method is used to specify how the ASP.NET application will respond to HTTP requests. The request pipeline is configured by adding [middleware](middleware.md) components to an `IApplicationBuilder` instance that is provided by dependency injection.
 
-Your `Configure` method must accept an `IApplicationBuilder` parameter. Additional services, like `IHostingEnvironment` and `ILoggerFactory` may also be specified, in which case these services will be [injected](dependency-injection.md) by the server if they are available. In the following example from the default web site template, you can see several extension methods are used to configure the pipeline with support for [BrowserLink](http://www.asp.net/visual-studio/overview/2013/using-browser-link), error pages, static files, ASP.NET MVC, and Identity.
+In the following example from the default web site template, several extension methods are used to configure the pipeline with support for [BrowserLink](http://vswebessentials.com/features/browserlink), error pages, static files, ASP.NET MVC, and Identity.
 
-[!code-csharp[Main](../common/samples/WebApplication1/Startup.cs?highlight=8,9,10,14,17,19,23&start=58&end=86)]
+[!code-csharp[Main](../common/samples/WebApplication1/Startup.cs?highlight=8,9,10,14,17,19,21&start=58&end=86)]
```

Comment:
>can you replace start/end with snippet?

This article uses `/common/samples/WebApplication1/` -- if I change that code I change line numbering for a dozen other articles.  Suggest we address snippetization in project and articles when we update the project for csproj.